validation/src/test/scala/hmda/validation/rules/lar/quality/Q030Spec.scala

```diff
@@ -1,210 +1,33 @@
-              smallCountyOutsideMSA.withoutTract.copy(state = someOtherState).mustFail(craOrNot)
-            }
-          }
-          "county is large" must {
-            "fail" in {
-              largeCountyOutsideMSA.withoutTract.mustFail(craOrNot)
-            }
-          }
-        }
-        "tract is present" when {
-          "state, county, and tract match" must {
-            "pass" in {
-              largeCountyOutsideMSA.mustPass(craOrNot)
-            }
-          }
-          "state, county, and tract do not match" must {
-            "fail" in {
-              smallCountyOutsideMSA.copy(state = someOtherState).mustFail(craOrNot)
-              smallCountyOutsideMSA.copy(tract = someOtherTract).mustFail(craOrNot)
-            }
-          }
-        }
-      }
-
-      {
-        val msaOrNot = Gen.oneOf("13820", "NA")
-        val tractOrNot = Gen.oneOf("0304.08", "NA")
-
-        "state is present but county is NA" must {
-          forAll(craOrNot) { implicit fi: Institution =>
-            s"fail when CRA is ${fi.cra}" in {
-              forAll(msaOrNot, tractOrNot) { (msa, tract) =>
-                Geography(msa, "01", "NA", tract).mustFail
-              }
-            }
-          }
-        }
-        "county is present but state is NA" must {
-          forAll(craOrNot) { implicit fi: Institution =>
-            s"fail when CRA is ${fi.cra}" in {
-              forAll(msaOrNot, tractOrNot) { (msa, tract) =>
-                Geography(msa, "NA", "117", tract).mustFail
-              }
-            }
-          }
-        }
-      }
-
-      implicit class GeoChecker(geo: Geography) {
-        def mustFail(implicit fi: Institution) = forAll(larGen) { lar => lar.copy(geography = geo).mustFail }
-        def mustPass(implicit fi: Institution) = forAll(larGen) { lar => lar.copy(geography = geo).mustPass }
-        def withoutTract = geo.copy(tract = "NA")
-        def mustFail(fiGen: TableFor1[Institution]): Assertion = forAll(fiGen) { fi => geo.mustFail(fi) }
-        def mustPass(fiGen: TableFor1[Institution]): Assertion = forAll(fiGen) { fi => geo.mustPass(fi) }
-      }
+  property("passes when action taken is in range and geography is not NA") {
+    forAll(larGen, Gen.choose(1, 6), stringOfN(3, Gen.alphaChar)) { (lar, i, geo) =>
+      val validGeo = Geography(geo, geo, geo, geo)
+      val validLar = lar.copy(actionTakenType = i, geography = validGeo)
+      validLar.mustPass
     }
   }
 
-  implicit class LarChecker(lar: LoanApplicationRegister) {
-    def mustFail(implicit fi: Institution) = check(lar, fi) mustBe a[Failure]
-    def mustPass(implicit fi: Institution) = check(lar, fi) mustBe a[Success]
-    def check(lar: LoanApplicationRegister, fi: Institution) = Q030.inContext(ValidationContext(Some(fi), None)).apply(lar)
+  property("fails when action taken is in range and geography is NA") {
+    forAll(larGen, Gen.choose(1, 6)) { (lar, i) =>
+      val invalidGeo = Geography("NA", "NA", "NA", "NA")
```

Comment:
We might want a spec or two that shows what happens when only one, two, or three of the Geography fields is "NA".